generator/config/schema.json

```diff
@@ -214,8 +253,123 @@
                     "items": {
                         "type": "object"
                     }
+                },
+                "schema": {
+                    "type": [
+                        "string",
+                        "object"
+                    ],
+                    "additionalProperties": {
+                        "type": "object",
+                        "additionalProperties": {
+                            "$ref": "#/definitions/SimplifiedSchemaField"
+                        }
+                    }
                 }
             }
+        },
+        "SimplifiedSchemaType": {
+            "anyOf": [
+                {
+                    "$ref": "#/definitions/SimplifiedSchemaBaseType"
+                },
+                {
+                    "$ref": "#/definitions/SimplifiedSchemaArrayType"
+                },
+                {
+                    "$ref": "#/definitions/SimplifiedSchemaDocumentType"
+                }
+            ]
+        },
+        "SimplifiedSchemaField": {
+            "additionalProperties": false,
+            "properties": {
+                "types": {
+                    "items": {
+                        "$ref": "#/definitions/SimplifiedSchemaType"
+                    },
+                    "type": "array"
+                }
+            },
+            "required": [
+                "types"
+            ],
+            "type": "object"
+        },
+        "SimplifiedSchemaArrayType": {
+            "additionalProperties": false,
+            "properties": {
+                "bsonType": {
+                    "const": "Array",
+                    "type": "string"
+                },
+                "types": {
+                    "items": {
+                        "type": "object"
+                    },
+                    "type": "array"
+                }
+            },
+            "required": [
+                "bsonType",
+                "types"
+            ],
+            "type": "object"
+        },
+        "SimplifiedSchemaBaseType": {
+            "additionalProperties": false,
+            "properties": {
+                "bsonType": {
+                    "enum": [
+                        "Array",
+                        "Binary",
+                        "Boolean",
+                        "Code",
+                        "CodeWScope",
+                        "Date",
+                        "Decimal128",
+                        "Double",
+                        "Int32",
+                        "Int64",
+                        "MaxKey",
+                        "MinKey",
+                        "Null",
+                        "ObjectId",
+                        "BSONRegExp",
+                        "String",
+                        "BSONSymbol",
+                        "Timestamp",
+                        "Undefined",
+                        "Document",
+                        "Number"
+                    ],
```

Comment:
We'll have to converge on the same type names for operator return types and argument types. I'm fine with changing the names in [`expressions.php`](https://github.com/mongodb/mongo-php-library/blob/ae6a3a53d507fe38e447dfde4e77c8efe5724925/generator/config/expressions.php#L19-L38) to be consistent with other tools.